tests/core/data/test_csv_data_node.py

```diff
@@ -429,3 +430,29 @@ def check_data_is_positive(upload_path, upload_data):
 
         # The upload should succeed when check_data_is_positive() return True
         assert dn._upload(new_csv_path, upload_checker=check_data_is_positive)
+
+    def test_duplicate_data_file(self):
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/example.csv")
+        dn = CSVDataNode("foo", Scope.SCENARIO, properties={"path": path, "exposed_type": "pandas"})
+        _DataManager._set(dn)
+
+        read_data = dn.read()
+        assert read_data is not None
+
+        old_path = dn.path
+        new_file_path = str(dn._duplicate_data())
+        assert filecmp.cmp(path, new_file_path)
```

Comment:
What about adding an explicit assert to ensure the new path is correct?
```suggestion
        assert new_file_path == "data_sample/TAIPY_DUPLICATED_example.csv"
        assert filecmp.cmp(path, new_file_path)
```